src/route/costing/mod.rs

```diff
@@ -0,0 +1,97 @@
+pub mod auto;
+pub mod bicycle;
+pub mod bikeshare;
+pub mod bus;
+pub mod hov;
+pub mod motor;
+pub mod motorcycle;
+pub mod multimodal;
+pub mod pedestrian;
+pub mod taxi;
+pub mod truck;
+
+use serde::Serialize;
+
+#[derive(Serialize, Clone, Debug)]
+#[serde(tag = "costing", content = "costing_options")]
+#[allow(clippy::large_enum_variant)]
+pub enum Costing {
+    /// Standard costing for driving routes by car, motorcycle, truck, and so on.
+    ///
+    /// Obeys automobile driving rules, such as access and turn restrictions.
+    /// This provides a short time path (though not guaranteed to be the shortest time) and
+    /// uses intersection costing to minimize turns and maneuvers or road name changes.
+    /// Routes also tend to favor highways and higher classification roads,
+    /// such as motorways and trunks.
+    #[serde(rename = "auto")]
+    Auto(auto::AutoCostingOptions),
```

Comment:
It doesn't seem particularly useful to add these stubs here if they're empty. Can these just be unit enums unless the structs are non-empty?

Reply:
The stubs here are mostle there because I decided to push the related changes to another PR (for reviewability).

Changing from an unit enum,unit struct,.. to a struct is semver breaking, as it now has different characterisitcs.
=> I'd prefer to keep it as is.

Also this way, others know where to add the costing options..

Reply:
I actually think that's fine - we can do an a semver bump if necessary once we get there. If you're planning to add these in another PR, that would probably be before a release (and the next semver bump) anyway. 

Adding these without actual fields requires users to deal with a more complicated API (``Cycle(Default::default)``) vs (``Cycle``) without any actual benefits.

Anyway, I don't feel super strongly about this so let's merge as-is (rather than making you refactor) but in general would like to prevent stubs that don't actually provide any value.